### ISSUE TITLE
Can't connect to whatsapp

### ISSUE THREAD

**Issue by codingr** [davidgfnet/whatsapp-purple#224]
"xxxxxxxxxx disconnected
Server closed the connection"

This message appears when I start pidgin. It worked perfectly 24 hours ago. I have tried setting server as c.whatsapp.net, c2.whatsapp.net, c3.whatsapp.net, changing port to 443 and 5222, resource Android-2.11.151-443 to Android-2.31.151-443... with no luck.

I've read some spanish guys who are suffering the same problem.My OS is Windows 7 with SP1, just in case.

**Comment by nicoceledon** [comment]
I've got the same problem, but i just tried with c3.whatsapp.net and c2.whatsapp.net, with port 443 and Android-2.31.151-443. 

I'm using debian Jessie.

**Comment by KottV** [comment]
the same :(

**Comment by annonbacca** [comment]
Happen with me too. But I think its probably an auto blocking mechanism they are using.

**Comment by vanfanel** [comment]
Same here. Tried the different servers and ports with no luck.
Are they locking us out?

(OS is not related to this issue as far as I understand, but I'm on Debian GNU/Linux)

**Comment by davidgfnet** [comment]
it works for me... Tried an hour ago and I could see all the offline
messages people sent me.

**Comment by vanfanel** [comment]
@davidgfnet: in the first place, a HUGE thanks for this great piece of software! :)

Can you please post the exact settings you are using? (server, port, android version...)
Maybe they are locking users progressively?
Using the latest github code here with no luck since yesterday.

**Comment by davidgfnet** [comment]
I'll do as soon as I get home. But I don't think it matters at all... I've
been changing them and using my phone to debug and implement new features
and it always works...
Maybe (just maybe) they know when and how people activated they numbers and
block using that as discriminant. My number is really old, I activated it
using a very old version of yowsup (maybe 1 year or more).

**Comment by vanfanel** [comment]
@davidgfnet : My number is also very old and I registered via yowsup too, a year ago. So we are in the same situation.

**Comment by davidgfnet** [comment]
Not a US number for sure

**Comment by vanfanel** [comment]
Mine's a normal spanish number. 34 610 XXX XXX, not an US number.

**Comment by Burglory** [comment]
I cannot connect either.
I am using pidgin:
2.10.9.
c2.whatsapp.net
443
Android-2.31.151-443

I have a dutch number. I registered three years ago via yowsup and reregistered 1 1/2 years ago. The server closes the connection. This started this weekend.
I am using whatsapp dll version from 6 september 2014 (Version 0.1).

**Comment by annonbacca** [comment]
@vanfanel 

most likely they are locking us out. Had the problem ever since I intentionally restarted  it while switching ip address. Most likely its a block, my only hope its a temporary thing cause I have heard users of an alternate client "whatsapp plus" were getting blocked like randomly. So its probable they are handing out blocks based on our strange whatsapp app version numbers. Additionally at the time, I was using c2.whatsapp.net while the other numbers were using c3.whatsapp.net

I registered using a rather old version of WART, and in the time I registered this number of mine that got blocked, I registered for one other number, and that number has typically been dormant, so when this number got blocked, I wasn't using the other number, and I am still able to use that number along with other numbers which I have created over time, which were made with varying versions of WART but similar whatsapp android version. I think the real issue is the blocking.

**Comment by davidgfnet** [comment]
Yep,
Let me add some verbose output (so we can know the reason why loging in
fails) and what is happening.

**Comment by rutja76** [comment]
Here's an extract from my pidgin log (I've X:ed some information):

(16:42:35) account: Connecting to account XXXXXXXX.
(16:42:35) connection: Connecting. gc = XXXXXXX
(16:42:35) GdkPixbuf: gdk_pixbuf_composite: assertion `GDK_IS_PIXBUF (dest)' failed
(16:42:35) GLib-GObject: g_object_unref: assertion `G_IS_OBJECT (object)' failed
(16:42:35) waprpl: logging in XXXXXXXXXXXXX
(16:42:35) dnsquery: Performing DNS lookup for c2.whatsapp.net
(16:42:35) dnsquery: IP resolved for c2.whatsapp.net
(16:42:35) proxy: Attempting connection to 184.173.147.53
(16:42:35) proxy: Connecting to c2.whatsapp.net:433 with no proxy
(16:42:35) proxy: Connection in progress
(16:42:40) util: Writing file prefs.xml to directory C:\Program Files (x86)\PidginPortable\.purple
(16:42:40) util: Writing file C:\Program Files (x86)\PidginPortable\.purple\prefs.xml
(16:42:40) util: Writing file accounts.xml to directory C:\Program Files (x86)\PidginPortable\.purple
(16:42:40) util: Writing file C:\Program Files (x86)\PidginPortable\.purple\accounts.xml
(16:42:41) util: Writing file blist.xml to directory C:\Program Files (x86)\PidginPortable\.purple
(16:42:41) util: Writing file C:\Program Files (x86)\PidginPortable\.purple\blist.xml
(16:42:41) accels: saving accels to C:\Program Files (x86)\PidginPortable\.purple\accels
(16:42:56) proxy: Connecting to c2.whatsapp.net:433.
(16:42:56) proxy: Error connecting to c2.whatsapp.net:433 (Connection timed out.).
(16:42:56) proxy: Connection attempt failed: Connection timed out.
(16:42:56) connection: Connection error on XXXXXXX (reason: 0 description: Unable to connect: Connection timed out.)
(16:42:56) GdkPixbuf: gdk_pixbuf_composite: assertion `GDK_IS_PIXBUF (dest)' failed
(16:42:56) GLib-GObject: g_object_unref: assertion `G_IS_OBJECT (object)' failed
(16:42:56) account: Disconnecting account XXXXXXXXXXXXXX (XXXXXXXXX)
(16:42:56) connection: Disconnecting connection XXXXXXXX

**Comment by nicoceledon** [comment]
i've got this from the log

Tag: start
at[from]=s.whatsapp.net
Data:  ...

Tag: stream:features
Data:  ...

Tag: challenge
Data: ËÊù•66¼qJ ...

Tag: response
Data: M(ÀÙCˆhq
 ...

Tag: failure
Data:  ...
 Tag: not-authorized
 Data:  ...

**Comment by davidgfnet** [comment]
So... It seems WA is blocking users. I recall there was some sort of
"blocked" comment on some errors (i think it was cause:blocked).
Have you guys tried to ask for a new password using WART or any tools alike?
If WART complains it is because your number is probably blocked.

**Comment by cassiodoroVicinetti** [comment]
I got the same log as nicoceledon, then I asked for a new password by WART and now it's working!

**Comment by davidgfnet** [comment]
Told yaaaaa!
Anyway, nice to hear it works for you guys.

**Comment by vanfanel** [comment]
Hmmm.... so asking for a new password fixes it?
Is there a way to receive the new password without a phone? I haven't the number I use with this lib anymore.

**Comment by nicoceledon** [comment]
That was all! Request the password and running.

@vanfanel  As for the password request without a phone, i don't think you can't because they send you the code to the phone.

**Comment by davidgfnet** [comment]
I'm using WP7-2.31.151-443 as resource connecting to c1.whatsapp.net
But I think the problem is the password, so... I think i will close this. So far recommentations are: as usuarl try a fresh install/config, try my resource, try different hostnames and if that does not work, ask for a new password.

Thanks guys!

**Comment by rutja76** [comment]
Changing the password does not seem to work for me: I get connection timed out (as reported above) on c1, c2 and c3.whatsapp.net

**Comment by Burglory** [comment]
New password with WART worked for me!

**Comment by davidgfnet** [comment]
By timed out what you mean? You are able to use it but at some point it disconnects or you never get the chance to use it at all?

**Comment by rutja76** [comment]
Yes, timed out, it started yesterday evening. Now, I have to admit that I don't know whether changing password with WART has nothing to do with that. But I've tried c1, c2 and c3 servers and I was timed out an all of them on port 443, which I've been using all the time without any problems until yesterday evening. After changing password with WART I was still timed out, so I tried changing port to 5222 as it was mentioned in the first post of this thread. Only at that point I was able to login on c3 server (haven't tried the other ones) using Android-2.31.151-443 as resource.

**Comment by davidgfnet** [comment]
Maybe whatsapp servers were down... just saying. I mean, behind any hostname (ie. c3) there are a bunch of servers (different IPs) so bad luck can make you ending up in a broken server. I guess the official app has a round robin mechanism so everytime there is a timeout it retries a different IP. (seems that pidgin does not do that).
On top of that they are kicking people using non-official clients such as WA+, that could be another reason... I don't know why people is so dumb not to migrate to Telegram.

**Comment by rutja76** [comment]
Actually port 443 (on c1, c2 and c3) doesn't work for me anymore since yesterday evening, and I have been using it for a year. Just retried: timed out. Only port 5222 seems to work.

**Comment by KottV** [comment]
i've tried openwapp (in firefox os emul), and it worked
then, re-registered with yowsup, and whatsapp-purple connected flawlessly...

**Comment by mlopezgva** [comment]
I have same prolem here since 04:00 GMT (approx.), but can't run yowsup in
this machine, I'll have to wait 'to return home. :(

2015-02-24 22:34 GMT-03:00 Konstantin <notifications@github.com>:

> i've tried openwapp (in firefox os emul), and it worked
> then, re-registered with yowsup, and whatsapp-purple connected
> flawlessly...
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/davidgfnet/whatsapp-purple/issues/224#issuecomment-75888837>
> .
>



-- 
Atte.,

*Manuel López*

**Comment by davidgfnet** [comment]
Use WART instead (it's a Windows Application), you only need .NET I think

**Comment by dequis** [comment]
@davidfraser you might want to keep this one open so people don't open new ones

(they will do anyway, but hopefully less)

**Comment by davidgfnet** [comment]
Yeah, thanks dx, what a bunch of stupid morons.

**Comment by mlopezgva** [comment]
Last WART version requests a PIN via SMS, and I have currently no access to
this line. I've tried with an older version, but it throws and error. :/

2015-02-25 9:58 GMT-03:00 David G. F. <notifications@github.com>:

> Use WART instead (it's a Windows Application), you only need .NET I think
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/davidgfnet/whatsapp-purple/issues/224#issuecomment-75955267>
> .
>



-- 
Atte.,

*Manuel López*

**Comment by davidgfnet** [comment]
Ehm Manuel, you'll need the phone number. You just cannot get a new
password without that SMS. I mean, it is impossible otherwise.

**Comment by mlopezgva** [comment]
hmpf... I'll have to borrow a terminal, since mine is broken... Before that
I've never needed the phone, just my number. :(

thanks for the response, nonetheless. I'll try later to get my registration
using yowsup.

2015-02-25 10:44 GMT-03:00 David G. F. <notifications@github.com>:

> Ehm Manuel, you'll need the phone number. You just cannot get a new
> password without that SMS. I mean, it is impossible otherwise.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/davidgfnet/whatsapp-purple/issues/224#issuecomment-75961300>
> .
>



-- 
Atte.,

*Manuel López*

**Comment by davidgfnet** [comment]
I seriously doubt that assertion. You always need a phone to verify that you own that number, otherwise I could use your phone.

**Comment by jbaroli** [comment]
I have the same situation here, using 1.1.4 and then installed 1.1.7 and it´s the same.
I registered my account with an Android phone and it works OK, so it looks like sth with whatsup trying to connect to whatsapp servers.

**Comment by codingr** [comment]
Changing the password worked for me. Thanks.

**Comment by salvattore001** [comment]
i have just installed the plugin for the first time. got a password from WART. tried different hostnames. always get the error "server closed the connection". what else could i try?

**Comment by davidgfnet** [comment]
Make sure the phone number is the "correct" one. Many people get it wrong.

**Comment by salvattore001** [comment]
i got an SMS from WART. so it is supposed to be the correct number? in case.. how i know which is the correct number?

**Comment by davidgfnet** [comment]
I mean make sure to write the number in the right format. International prefixes may be written as +XX or 00XX for instance.

**Comment by salvattore001** [comment]
tried 54 and +54 still no luck :( the number is from argentina

**Comment by seenutn** [comment]
I tried to register again with yowsup-cli I had earlier and yowsup-cli itself failed (The issue I encountered was same as https://github.com/tgalal/yowsup/issues/353).  I downloaded the fresh yowsup-cli from github, requested the password and with that password the pidgin is working properly with whatsapp.  Thanks to tglal and davidgfnet.

**Comment by dequis** [comment]
>tried 54 and +54 still no luck :( the number is from argentina

Also try 549 (as in 54911... instead of 5411...) . I have no idea why it needs a 9 sometimes, but it does.

**Comment by davidgfnet** [comment]
Luckily for you there are some Argentinean users :D
I just committed a change which should help debugging. It should clearly state "not-authorized" in case the password (or the phone) is incorrect.

**Comment by salvattore001** [comment]
adding the 9 worked! 
it seems that the 9 means it is a cell phone.. thanks!

**Comment by githubtefo** [comment]
We are from Argentina too.
Since 4 days ago my girlfriend's account has the message "Server closed the connection". We tried to put "00", "+" with/without the "'9" with no results. Even we changed the Resource and nothing.
She is living in another country so we cannot receive the sms code for restarting the account. Is there any other way to solve this? She is having problems to contact with her family and friends because all people use this f%&"¬ protocol! I hate The Corporation. Sorry but I'm so angry. Thanks david for your fight on the implementation of open communication protocols. C'mon, this is XMPP! Thanks.

**Comment by davidgfnet** [comment]
Hold on, if you have the SIM no matter in which country you are, you can still receive the SMS. And I'm pretty sure whatsapp forced a password update or something...

**Comment by eksperimental** [comment]
@davidgfnet how do i get a more verbose output on linux?
i' m using  `pidgin -d > ~/debug.log 2>&1`

thanks

**Comment by davidgfnet** [comment]
If you want to get a full WA protocol trace on linux you can compile the
plugin using "make debug".
This will print loads of stuff on stdout.

**Comment by eksperimental** [comment]
@davidgfnet thanks

**Comment by TerrificTT** [comment]
Firstly kudos for such a great software which has helped me a lot.
But as of late i'm stuck with this message "Server closed the connection".
Tried all the tweaks but no progress.

**Comment by githubtefo** [comment]
Thanks david for yor answer!
But for some reason she cannot use her argentinian number in Brasil with roaming. Useless mobile companies...
Now she buy a Firefox OS smartphone with a brasilian number and use ConectA2. So I'd like to ask you if it is possible to open the account with 2 different devices and applications, or whatsapp network will kick her account?
Thanks guys!

**Comment by axel21a** [comment]
@TerrificTT look here #178

**Comment by davidgfnet** [comment]
Only one device connected at a time. Whatsapp rules.

**Comment by TerrificTT** [comment]
Debugging pidgin gave this and there seems to be nothing supernatural out here...then what may be the cause...(I have changed the password 20th time still it gives this response)
(14:03:05) account: Connecting to account XXXXXXXXXXXX.
(14:03:05) connection: Connecting. gc = XXXXXXXXXXXX
(14:03:05) whatsapp: logging in XXXXXXXXXXXX
(14:03:05) dnsquery: Performing DNS lookup for c3.whatsapp.net
(14:03:05) autorecon: done calling purple_account_connect
(14:03:05) dns: DNS child 11358 no longer exists
(14:03:05) dns: Created new DNS child 11366, there are now 1 children.
(14:03:05) dns: Successfully sent DNS request to child 11366
(14:03:05) dns: Got response for 'c3.whatsapp.net'
(14:03:05) dnsquery: IP resolved for c3.whatsapp.net
(14:03:05) proxy: Attempting connection to 50.22.225.69
(14:03:05) proxy: Connecting to c3.whatsapp.net:5222 with no proxy
(14:03:05) proxy: Connection in progress
(14:03:05) proxy: Connecting to c3.whatsapp.net:5222.
(14:03:05) proxy: Connected to c3.whatsapp.net:5222.
(14:03:09) connection: Connection error on XXXXXXXXXXXX (reason: 0 description: Server closed the connection)
(14:03:09) account: Disconnecting account XXXXXXXXXXXX (XXXXXXXXXXXX)
(14:03:09) connection: Disconnecting connection XXXXXXXXXXXX
(14:03:09) connection: Destroying connection XXXXXXXXXXXX
(14:03:14) util: Writing file accounts.xml to directory /home/t/.purple
(14:03:14) util: Writing file /home/t/.purple/accounts.xml
dns[11366]: nobody needs me... =(

**Comment by lovetox** [comment]
so tried to request a new password with WART but it says "fail, reason: blocked"

so does this mean i can never use whatsapp again with that number ?!

**Comment by joshuakraemer** [comment]
I have the same problem. I have sent a new request with yowsup, but get "status: fail, reason: blocked".

**Comment by davidgfnet** [comment]
In that case the only option left is begging the whatsapp team. It worked for me with two numbers. Also another one got unblocked after some time (I think it might be weeks).

**Comment by CleberMetal** [comment]
Hi,
I have the same problem than joshuakraemer. When I sent new request using WART-1.7.3.0, I got the answer: "status: fail, reason: blocked".
The news: I tried too from a new mobile number that I never had used before, and I got "SMS Sent". Is it possible the WhatsApp server's blocked my mobile numbers? Do you have some tip?
Tks

**Comment by dequis** [comment]
Yes, it means exactly that, your number is blocked. And like david said, "the only option left is begging the whatsapp team". Go send them an email to the support address.

**Comment by joshuakraemer** [comment]
Let me add this.  I registered my number for the first time with yowsup, then used it for some regular chatting with a few friends for one week, then my account was blocked.  The reason for blocking may be that I registered the number with yowsup and not the official client.  I think it is advisable to add a hint in the readme that this might happen.

**Comment by dequis** [comment]
No, registering through yowsup should be perfectly okay. Pretty much everyone registered their numbers with third party apps, yowsup is no different from the rest.

**Comment by rutja76** [comment]
I got my number blocked as well. Cannot login even after installing the official app on my mobile: "Your number is no longer allowed to use our service" (http://www.whatsapp.com/faq/en/general/23154266). I guess they are kicking out people who use whatsapp through third-party applications. As written in the above linked FAQs, "Using an automated system or an unauthorized / unofficial client application to send messages through WhatsApp" is in violation of their terms of service.

**Comment by davidgfnet** [comment]
I agree with dx, all activators are the same those days (exact same implementation on most of them).
Maybe I'm really lucky :D

**Comment by Jeronimo17** [comment]
I also have been banned 2 days ago, I'm using the user-agent S40-2.12.67 I check from time to time here http://www.whatsapp.com/s40/OptWhatsApp.jad.

I do not upgrade in time on this occasion, it may be that the cause of the ban or that I have sent pictures and always the same thumbnail is sent...

Should be checked version of the user-agent to connect

**Comment by davidgfnet** [comment]
Oh now that you mention! The thumbnail! It is a very easy way to detect this particular plugin.
I think I'll fix this ASAP. Profile picture is another thing we should get fixed too.
Thanks for the heads up

**Comment by Cybernetic1** [comment]
Does it mean that if my number is blocked by Whatsapp, there's no way to revive it?  I e-mailed them and they simply pointed to this answer :(

**Comment by annonbacca** [comment]
Just got another number blocked. Feeling miserable cause I can't try reactivating the number via SMS, lost the SIM card. Anyways this happened to me on a number registered using the latest WART and has only been used on pidgin.

I think, the blocking mechanism takes place during logging on stage (Connection to server) so if a number has been running for weeks on end, non-stop with continuous internet connection, you are less likely to be banned whereas those that happen to run it on a choppy bad internet connection with lots of disconnects would face eventual block. Like the checking mechanism for fake client, I suspect runs on a random basis, so its possible to connect and disconnect a couple of times to the hour but eventually a trace would happen and the number could get blocked.

I am not sure as for the latest whatsapp, but I was running mine using one of the old dlls which doesn't have any of the read receipt capability. So far, my other accounts which were registered using the official client are still running, and I do switch them on pidgin from time to time, and so far no problem.

And yes, the ones made using the official client have profile picture and status messages adjusted so they don't just appear as blank profile with an equally blank status or the default, "Hey, I am on whatsapp" message.

The other final thing I would like to add is that the WART client uses an unknown fixed google account (or probably none at all) while the whatsapp official for android has a tie in with a unique google account (which is used mainly for push notification), its how they know if the same user has resurrected a deleted account on the same number, in addition to the user identifiable information such as  phone contacts and IMEI + MAC. I suppose the iOS based Apple phones has a tie in with iCloud. So, I suppose it won't be hard for them to eventually spot all those that do not have a distinct google/iCloud account, or all those that are registered to a single account.

Finally, I would like to take this opportunity to thank DavidFG for creating a fantastic software that allows me to use Whatsapp on PC. Thank you man, most appreciate it.

**Comment by davidgfnet** [comment]
I'm kinda lazy those days but in the past I used to implement everything by capturing real Wa traffic from my phone. So... Pcap traces are more than welcome!

**Comment by davidgfnet** [comment]
Oh btw. Which is the best method to extract wa password from an android phone? I may give that i try.

**Comment by dequis** [comment]
https://github.com/mgp25/WhatsAPI-Official/wiki/Extracting-password-from-device

**Comment by davidgfnet** [comment]
Yeah. No way I'm running an apk without compiling it myself, specially when it requires root access.. Not that crazy.
Thanks anyway. I might decompile it and give it a look

**Comment by Red-Hell** [comment]
Same problem here. Account banned yesterday morning. I've also tried installing whatsapp under android emulator and says that my number is banned.

I hope they will unban my account...

**Comment by dequis** [comment]
@davidgfnet the 'bash' section in the same page provides a do-it-yourself method

BTW the situation in this ticket is looking pretty bad. Maybe creating a 'whatsapp web' client isn't a bad idea after all... I wrote [some notes][1] about it, only good enough as an introduction/overview, but there's plenty of stuff left to reverse to achieve a working third party client for that. It's also completely different from this plugin... And I mentioned it there, but there's a restriction to only one web client at the same time (the others get in a pause-like mode, and gotta implement a way to takeover the mutex)

[1]: http://wiki.dequis.org/notes/whatsapp_web/

**Comment by davidgfnet** [comment]
Yeah thanks, I managed to get my pwd but it's very tricky to eavesdrop it since the client usually reconnects using the simplified handshake mechanism instead of doing a full handshake. I haven't still figured out whether we can decrypt that.
What you say sounds interesting but IMO this defies the whole purpose of all of this which is getting rid of the official client. The main consumers of this plugin seem to be people not having an smartphone at all.

**Comment by Red-Hell** [comment]
In fact, David, I haven't a smartphone ad I don't want it (I don't want to pay for another internet connection ad I can't use something touch, I prefer keys).

But David, you have to be proud of your plugin, also if it lacks of something, if the only way that the WA team has to block other program is blocking the account.

In another system you will have only to register again, but change the phone number is not really easy...

**Comment by Red-Hell** [comment]
"Hi,

Your WhatsApp account has been banned because your activity on WhatsApp is in violation of our Terms of Service.

We cannot comment on the individual reasons that have led to your account ban. However, the following actions provide examples of activity that will result in an account becoming banned:

bla bla as written in faq

Please review our Terms of Service to learn more about the appropriate uses of WhatsApp.

We currently do not have a repeal process in place for unbanning your account."

Now I really doesn't need a smartphone...

Sorry David, but you will have to write this on the homepage of this projet, to use this plugin at risk of ban...

**Comment by davidgfnet** [comment]
Well... People usually accept the ToS without reading them. I agree I could explicitly warn people but come on, they are grown up, it's time to assume responsibilities.

**Comment by Jeronimo17** [comment]
You're doing a good job, you're helping people to switch to Telegram :P

**Comment by assegaf** [comment]
@davidgfnet did you managed to able to see live debug log IN/OUT of whatsapp official client ?
how to do this in simple way ? 
maybe something mismatch different mechanism between official and this api. thanks

**Comment by davidgfnet** [comment]
Extract your pass. Make sure WA connects using a full handsake (see my net-tools repo). Use wireshark with the freshly extracted password. Make sure to kill the app (just reboot the phone to make sure) and start using watsapp on your phone.

**Comment by assegaf** [comment]
ah yes, sorry, reboot the phone to get Full handshake, I will try now ..

**Comment by namoplasta** [comment]
@davidgfnet 
When installed whatsapp-purple from the source code, is still maintained Android-2.11.151-443, and only works on Android-2.31.151-443, I think this must be updated in git.
Only worked with me at the port 5222, think it should be a warning in the plugin to try this port if the 433 does not work.

**Comment by marga-debian** [comment]
I got one account blocked back by the end of February, I wrote to request an unblock and they replied saying that I was using an unauthorized program and that they wouldn't unblock.

I'm now having a somewhat similar but different problem with a different number. It frequently disconnects with the "Server closed connection" message, but then it reconnects fine. It just doesn't stay connected long, sometimes it's seconds, sometimes it's minutes.  Sometimes after reconnecting I get the backlog, sometimes I don't. :-/

I've tried changing servers and resource, but that doesn't seem to make any difference.

Any tips on how to get a more stable connection?

**Comment by davidgfnet** [comment]
See #263

**Comment by gabriel-tandil** [comment]
My contribution:
After Whatsapp lock my account, and after pray to whatsapp support to unlock my numbers, with negative response on several occasions. Wait several weeks, I went back to try and my numbers works again.

**Comment by namoplasta** [comment]
Fuck WhatsAPP, Telegram rules.

**Comment by dektiram** [comment]
i use pidgin whatsapp for 2 moths, everything work well. But since 2015-07-21 my account was "not-authorized". I've try to change password using WART and I get new password. But it's not give solution, my account still not-authorized. I try to use new number and register it via WART, but the same problem is happend. Is that any change in whatsapp rule ?

**Comment by ikbalenjino** [comment]
now pidgin is end. good bye

**Comment by ViBENo1** [comment]
i'm tried to setup wa plugin but i guess i did something wrong or just impossible to make it work. i have a Nokia N9 with Whatsup. my number is already registered and it cannot connect with my phone if i authorise my number with the wa plugin and reverse. how can i setup both client to not de-authorise each other time when i connect?

**Comment by rodpmaia** [comment]
Worked for me using this:
c3.whatsapp.net
5222
Android-2.31.151-443

rutja76 commented on 24 Feb

Yes, timed out, it started yesterday evening. Now, I have to admit that I don't know whether changing password with WART has nothing to do with that. But I've tried c1, c2 and c3 servers and I was timed out an all of them on port 443, which I've been using all the time without any problems until yesterday evening. After changing password with WART I was still timed out, so I tried changing port to 5222 as it was mentioned in the first post of this thread. Only at that point I was able to login on c3 server (haven't tried the other ones) using Android-2.31.151-443 as resource.

PS: for timed out I mean "Connection timed out." I put an extract of the log here above.
